scripts/qa_crowbarsetup.sh

```diff
@@ -4491,7 +4491,7 @@ function onadmin_reapply_openstack_proposals
 function onadmin_upgrade_prechecks
 {
     if ! crowbarctl upgrade prechecks --format json | rubyjsonparse \
-        'exit false if j.keys.any? { |key| j[key]["required"] && !j[key]["passed"]}'
+        "exit false if j.any? { |test| test['required'] && !test['passed'] }"
```

Comment:
i have quickly tested it and it seems that the `!` gets interpreted by the shell rather than by ruby. 
it says `-bash: !test['passed']: event not found`

what works is:

```
"exit false if j.any? { |test| test['required'] &! test['passed'] }"
```
but i am not sure that this is correct. there is also:
```
"exit false if j.any? { |test| test['required'] && test['passed'] == false }"
```
which should be correct